competencies/templates/competencies_self_assessment.html

```diff
@@ -0,0 +1,91 @@
+{% extends "base.html" %}
+{% load static %}
+
+{% block content %}
+
+<div class="container-fluid">
+    <div class="row">
+        <div class="col">
+    
+            <div class="card">
+                <div class="card-body">
+                    <div class="card-title">
+                        <h4>Self-Assess Your Competencies</h4>
+                    </div>
+                    <div class="card-text">
+                        
+                        {% if competencies %}
+                        <form method="POST" class="w-100">
+                            {% csrf_token %}
+                            {{form.user.as_hidden}}
+                            {{form.is_visible|as_crispy_field}}
+                            {{ formset.management_form }}
+                            <div class="table-responsive">
+                                <table class="table table-sm mt-3" id="competenciesTable">
+                                    <thead>
+                                    <tr>
+                                        <th scope="col">Competency</th>
+                                        <th scope="col" class="text-center">No Knowledge</th>
+                                        <th scope="col" class="text-center">Want to Learn</th>
+                                        <th scope="col" class="text-center">Learning it now</th>
+                                        <th scope="col" class="text-center">Know it</th>
+                                    </tr>
+                                    </thead>
+                                    <tbody>
+                                        {% for form in formset %}
+                                        <tr>
+
+                                            <td>
+                                                {{form.id}}
+                                                {{form.user_assessment.as_hidden}}
+                                                {{form.rating.as_hidden}}
+                                                {{form.competency}}
+                                                
+                                            </td>
+                                            
+                                            <td class="text-center">
+                                                <div class="form-check form-check-inline">
+                                                    <label for="id_form-{{forloop.counter0}}-none"><i class="bi bi-square{% if form.rating.value == 'no_knowledge' %}-fill{% endif %}" title="I don't know anything about this competency"></i></label>
```

Comment:
nicely done with the IDs using forloop.counter! This catches a lot of people off-guard

Reply:
Cheers :smile: